File: Programmers/Lv0/등수매기기.py
def solution(scores):
    score = [(score[0] + score[1]) / 2 for score in scores]
    i = 1
    rank = [0] * len(score)

    while (0 in rank): #모든 점수에 랭크를 매길때 까지
        adj = 0 # 등수 조정값
        if score.count(max(score)) == 1: # 최고점이 한 명이라면
            rank[score.index(max(score))] = i # 최고점의 인덱스에 매칭되는 rank 인덱스에 등수 표기
            score[score.index(max(score))] = -1 # 데이터를 삭제하면 인덱스가 변경되므로 최고점을 없애버림
            i += 1
        else: # 최고점이 여러명이면
            max_indexes = [index for index, value in enumerate(score) if value == max(score)]
            # 최고점의 인덱스를 모음
            for idx in max_indexes:
                rank[idx] = i # 등수 기록
                score[idx] = -1 # 최고점 없애기
                adj += 1 # 공동 등수 조정값
        i += adj
    return rank

File: Programmers/Lv0/등수매기기.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

vector<int> solution(vector<vector<int>> scores) {
    vector<float> score;
    for (auto s : scores)
        score.push_back((float)(s[0] + s[1]) / 2);
    int i = 1, adj;
    vector<int> rank(score.size(), 0);

    while (find(rank.begin(), rank.end(), 0) != rank.end()) {
        adj = 0;
        float max_value = *max_element(score.begin(), score.end());
        printf("%f\n", max_value);
        if (count(score.begin(), score.end(), max_value) == 1) {
            rank[distance(score.begin(), find(score.begin(), score.end(), max_value))] = i;
            score[distance(score.begin(), find(score.begin(), score.end(), max_value))] = -1;
            i += 1;
        }
        else {
            vector<int> max_indexes;
            for (size_t idx = 0; idx < score.size(); ++idx)
                if (score[idx] == max_value)
                    max_indexes.push_back(idx); // 조건을 만족하는 인덱스 저장
            for (int idx : max_indexes) {
                rank[idx] = i;
                score[idx] = -1;
                ++adj;
            }
        }
        i += adj;
    }
    return rank;
}